src/main/logic/Player.cpp

```diff
@@ -51,3 +53,22 @@ Direction Player::isOutOfBounds(){
 	return (MobileEntity::shortToDirection(boundsBreach));
 }
 
+PlayerState Player::getState() const {
+	return curState;
+}
+
+void Player::hit(){
+	curState = PlayerState::HIT;
+	hitTick = curtick;
+}
+
+/* NEEDS TO BE REFACTORED LATER */
+void Player::tick(){
+	curtick++;
+	if(curState == PlayerState::HIT){
+		if(curtick >= (hitTick + 25)){
+			curState = PlayerState::DEAD;
+		}
+	}
+}
+/* END NEEDS TO BE REFACTORED LATER */
```

Comment:
I think we should roll #67 into this branch before we merge into develop. You are definitely going the right way, so we may as well go the full way. In Logic there is a double (which should be a long int) called gameTime, Player should hold a pointer to that and use that to time events. It is the count of milliseconds in game time, when the game pauses it pauses. Then with moving the lives inside the Player we will be completely OOP.